src/scicloj/kindly_render/notes/resource_loader.cljc

```diff
@@ -0,0 +1,31 @@
+(ns scicloj.kindly-render.notes.resource-loader
+  "Creates dynamic loaders that will insert <script src=...> if not already loaded"
+  (:require [clojure.string :as str]
+            [scicloj.kindly-render.shared.util :as util]
+            [scicloj.kindly-render.notes.resource-hiccup :as resource-hiccup]))
+
+(defmulti resource-loader :type)
+
+(defmethod resource-loader :js
+  [{:keys [src package]}]
+  [:script
+   (str/replace
+     "var loadedScripts = loadedScripts ?? {};  !loadedScripts[SRC] ? (loadedScripts[SRC] = true, document.head.appendChild(Object.assign(document.createElement('script'), { src: SRC, onload: () => console.log(`Script SRC loaded.`) }))) : console.log(`Script SRC is already loaded.`);"
```

Comment:
My implementation for this was far more complicated, because I had thee issue that the JS gets loaded async.
So I "saw" that the most simples aproach (simialr to yours), would "not yet" have the JS dependency loaded, when the first JS using it executed, so it failed.

I could only solve this via a more complicated way of using "promises"

Reply:
see here: https://github.com/scicloj/kindly-render/blob/51d86ed1dea91e56141b76df13025c7a52ab13ce/src/scicloj/kindly_render/note/to_hiccup_inline_js.cljc#L64

It does not work without the "promise" (or a similar approach)